js_modules/dagster-ui/packages/ui-core/src/pipelines/OpGraphSelectionInput.tsx

```diff
@@ -0,0 +1,53 @@
+import {useMemo} from 'react';
+import styled from 'styled-components';
+
+import {GraphQueryItem} from '../app/GraphQueryImpl';
+import {OpSelectionLexer} from '../op-selection/generated/OpSelectionLexer';
+import {OpSelectionParser} from '../op-selection/generated/OpSelectionParser';
+import {InputDiv, SelectionAutoCompleteInput} from '../selection/SelectionAutoCompleteInput';
+import {createSelectionLinter} from '../selection/createSelectionLinter';
+import {weakMapMemoize} from '../util/weakMapMemoize';
+
+export const OpGraphSelectionInput = ({
+  items,
+  value,
+  onChange,
+}: {
+  items: GraphQueryItem[];
+  value: string;
+  onChange: (value: string) => void;
+}) => {
+  const attributesMap = useMemo(() => {
+    const names = new Set<string>();
+    items.forEach((item) => {
+      names.add(item.name);
+    });
```

Comment:
⛳ 

```
const names = new Set(items.map(({name}) => name));
```

Reply:
I like having the for loop for the future in case we add more attributes and to be consistent with all the other inputs.